src/libraries/System.Net.Security/src/System/Net/Security/SniHelper.cs

```diff
@@ -16,12 +16,12 @@ internal class SniHelper
         private static readonly IdnMapping s_idnMapping = CreateIdnMapping();
         private static readonly Encoding s_encoding = CreateEncoding();
 
-        public static string GetServerName(byte[] clientHello)
+        public static string GetServerName(byte[] clientHello, int offset, int count)
```

Comment:
Should this just take a span?

Reply:
There is span overload just bellow. This is main entry for tests. I guess I can re-factor them and remove this completely.

Reply:
fixed.